src/layouting/transformation.jl

```diff
@@ -257,6 +257,52 @@ function apply_transform(f::PointTrans{N1}, point::Point{N2}) where {N1, N2}
     end
 end
 
+# The following methods are based on logic from the PROJ library,
+# which essentially try to find the true bounding box of the given
+# rectangle after transformation, by sampling from a regular grid
+# bounded by the rectangle in input space.
+
+function apply_transform(f::PointTrans{2}, rect::Rect{2, T}) where T
+    N = 21 # TODO: spin this into PointTrans or out into a method
+    umins = fill(T(Inf),  2)
+    umaxs = fill(T(-Inf), 2)
+    xmins = minimum(rect)
+    xmaxs = maximum(rect)
+    for x in range(xmins[1], xmaxs[2]; length = N)
+        for y in range(xmins[2], xmaxs[2]; length = N)
+            u = Makie.apply_transform(f, Point(x, y))
+            for i in 1:2
+                umins[i] = min(umins[i], u[i])
+                umaxs[i] = max(umaxs[i], u[i])
+            end
+        end
+    end
+
+    return Rect(Vec2(umins), Vec2(umaxs .- umins))
+end
+
+function apply_transform(f::PointTrans{3}, rect::Rect{3, T}) where T
+    N = _DEFAULT_RECT_TRANSFORM_DENSITY[]
```

Comment:
I think this global is gone? And we should add tests, since CI is green even though this should error!

Reply:
Oops...did not notice that.  I'll write some tests up tomorrow then.  Thanks for the review!